### PR TITLE
[fix bug 1396005] Firefox 57 Quantum page

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Add Firefox 57 Quantum preview page at `/firefox/quantum/`.
- ~Strings are not yet final, will update when everything is settled.~
- Page is behind the switch `57-quantum-preview` and will 404 until activated (due for the 26th).
- Page is due to be localized for `en-US` and `de` only.

## Issue / Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1396005

## Testing
Demo: 
https://www-demo3.allizom.org/en-US/firefox/quantum/
https://www-demo3.allizom.org/de/firefox/quantum/

### CODE REVIEW

**Comment by alexgibson** [comment]
Ok, [final copy is aproved](https://bugzilla.mozilla.org/show_bug.cgi?id=1396005#c27) and now all in place, so strings are frozen.

**Comment by peiying2** [comment]
@alexgibson we will need to review the template before we can extract the strings.

**Comment by peiying2** [comment]
@flodolo

**Comment by flodolo** [comment]
In the meantime, can you use `firefox/quantum.lang` as lang file? I need to test locally and see if there's anything strange.

**Comment by alexgibson** [comment]
> In the meantime, can you use firefox/quantum.lang as lang file? I need to test locally and see if there's anything strange.

Sure

**Comment by alexgibson** [comment]
> In the meantime, can you use firefox/quantum.lang as lang file? I need to test locally and see if there's anything strange.

You don't actually need me to specify this in the template, right?

**Comment by flodolo** [comment]
> You don't actually need me to specify this in the template, right?

yes, with add_lang_files

**Comment by flodolo** [comment]
> yes, with add_lang_files

Never mind, I completely misread the name of the template. It's already going to use firefox/quantum.lang by default.

**Comment by flodolo** [comment]
Can't spot anything unwrapped.

File: bedrock/firefox/templates/firefox/quantum.html [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+{% extends "firefox/base-pebbles.html" %}
+
+{% block page_css %}
+  {% stylesheet 'firefox-quantum' %}
+{% endblock %}
+
+{% block page_title %}{{ _('2x faster and 30% less memory') }} | {{ _('Firefox Quantum Browser') }}{% endblock %}
+{% block page_desc %}{{ _('The new Firefox arrives November 14, 2017') }}{% endblock %}
+
```
Comment (craigcook) [inline bedrock/firefox/templates/firefox/quantum.html]: Almost forgot, we should add the new meta page image here.

File: media/css/firefox/quantum.scss [hunk 2]
```diff
@@ -0,0 +1,998 @@
...
+            width: 25%;
+        }
+    }
+
+    @media #{$mq-desktop-wide} {
+        padding: 60px 0 280px;
+    }
+}
+
+@supports(display: grid) {
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply media/css/firefox/quantum.scss]: There's already a fallback to Flexbox for IE/Edge, so i think we should be good here. For old IE browsers I think the stacked column is acceptable enough.

File: bedrock/firefox/templates/firefox/quantum.html (hunk 1 above)
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/quantum.html]: Will do 👍

**Comment by alexgibson** [comment]
@craigcook added the meta image, and squashed. Thanks for thr review!

File: media/css/firefox/quantum.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (craigcook) [reply media/css/firefox/quantum.scss]: Ah, I was testing in IE10, which doesn't support flexbox. As long as it looks  good in current Windows browsers I think we're ok with the stacking as a fallback for the others.

**Comment by craigcook** [comment]
r+ but I think we should merge #5130 first

**Pull request merged**